### ISSUE TITLE
Narrator reads the NavigationViewItem when Visibility is Collapsed

### ISSUE THREAD

**Issue by brunasilvalopes** [microsoft/microsoft-ui-xaml#1034]
**Describe the bug**
On [Dynamics 365 Guides](https://www.microsoft.com/store/productId/9N038FB42KKB), we hide some NavigationViewItems until the user is logged in. When navigating with narrator through the NavigationView when some items are hidden, narrator still reads them instead of ignoring them.

**Steps to reproduce the bug**
1. Open Microsoft Dynamics 365 Guides app (https://www.microsoft.com/store/productId/9N038FB42KKB)
2. Turn on Narrator
3. Navigate to the hamburger menu
4. Press CAPS + Right - narrator will navigate to Home button
5. Press CAPS + Right - narrator focus disappear and narrator reads information about a button which visibility is set to Collapsed.
I was also able to repro the bug on XAML Controls Gallery with some coding involved, please see explanation on the bottom of this report

**Expected behavior**
Narrator shouldn't read elements which visibility is set to collapsed

**Screenshots**
![image](https://user-images.githubusercontent.com/13021878/61018079-680b7700-a34a-11e9-8945-26a10d436f3a.png)

**Version Info**
OS Version 18362.175

NuGet package version: 
Was able to repro this on 2 different versions of Microsoft.UI.Xaml:
2.1.190405004 and 2.2.190611001-prerelease

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| May 2019 Update (18362)            | Yes   |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :-------------------- | :------------------- |
| Desktop                 | Yes   |
| Mobile                   |    |
| Xbox                      |    |
| Surface Hub          |     |
| IoT                         |     |

**Additional context**
To repro this on XAML Controls Gallery, I had to add one line of code on the file NavigationRootPage.xaml.cs that hides the "All Controls" menu item on the left.
On the AddNavigationMenuItems method, I set the Visibility of _allControlsMenuItem to Collapsed, like so:
```
private void AddNavigationMenuItems()
        {
            foreach (var group in ControlInfoDataSource.Instance.Groups.OrderBy(i => i.Title))           
[...]
                if (group.UniqueId == "AllControls")
                {
                    this._allControlsMenuItem = item;
                    this._allControlsMenuItem.Visibility = Visibility.Collapsed; // THIS IS THE SINGLE LINE NEEDED TO REPRO BUG
                }
[...]
            }
```
I have screen recordings of this bug on both Guides and XAML Controls Gallery, I can provide that to you guys if needed.

**Comment by mrlacey** [comment]
Is there some wider guidance missing on collapsed items and accessibility support?

Is it possible to do something at a lower level (in a base control) to ensure any collapsed control is not exposed to automation/accessibility tooling?

**Comment by brunasilvalopes** [comment]
I don't know how to answer your first question, but about the second, I was reading online about this problem and apparently we are missing overriding the IsControlElementCore method on the automationPeer of the NavigationView: 
```
protected override bool IsControlElementCore()
     {
          return Owner.Visible != Visiblity.Collapsed;
     }
```

Source: https://thinkrethink.net/2019/02/11/wpf-narrator-reads-collapsed-content/

**Comment by jevansaks** [comment]
Thanks @brunasilvalopes ! I would have thought that the framework uses visibility to hide UIA elements, but maybe there's something odd for ListViewItems. If you want to take a crack at fixing this, we would welcome the contribution! :)

**Comment by brunasilvalopes** [comment]
Yes, I'll try to fix this when I have some free time.

**Comment by brunasilvalopes** [comment]
I've investigated this issue and realized it also happens with ListViewItems, which means this bug is on the XAML Framework. I tried overriding IsControlElementCore as I mentioned above, but that didn't solve the problem. 
There is one workaround, which is to set the AccessibilityView property of the hidden menu item to Raw. To do that, I binded the AccessibilityView property to the same property that Visibility was binded to, then I used a converter that converts a false Boolean into AccessibilityView.Raw:
```
<wxcontrols:NavigationViewItem
                    x:Uid="NavigationView_HomeMenuItem"
                    AutomationProperties.AccessibilityView="{x:Bind ViewModel.IsVisible, Mode=OneWay, Converter={StaticResource BooleanToAccessibilityViewConverter}}"
                    Visibility="{x:Bind ViewModel.IsVisible, Mode=OneWay}"
                    Tag="Home"  />

```

**Comment by jevansaks** [comment]
Thanks @brunasilvalopes ! This looks like a good workaround, I don't know why this isn't happening in the framework. We'll look post-WinUI 3.0.

For your workaround since you're using x:Bind you might consider [function bindings](https://docs.microsoft.com/en-us/windows/uwp/data-binding/function-bindings) instead of converters? They're a lot simpler to use/write.

**Comment by brunasilvalopes** [comment]
Thanks for the suggestion Jevan, I didn't know that function bindings existed!

**Comment by karenbtlai** [comment]
Does not repro with updated NavView so closing.
